### DISCUSSION TITLE
Realistische Koordinaten

### DISCUSSION THREAD

**Discussion by C1710** [marhei/TrainCompany-Data#64] [Ideas]
@marhei Kannst du mal die `Station.json` aus diesem Branch lokal ausprobieren?
https://github.com/C1710/TrainCompany-Data/tree/Map-Update

Ich habe da mal die Koordinaten (automatisiert) auf Basis der echten Längen- und Breitengrade angepasst. Der Maßstab ist weiterhin der gleiche (0-Punkt im Westen ist Düren, im Norden Altenbeken. Nürnberg hat die gleichen Koordinaten wie vorher auch).

So könnten zukünftig auch neue Bahnhöfe vollautomatisch platziert werden.

**Comment by C1710** [comment]
So hab ich das jetzt mithilfe von Scripts soweit automatisiert, dass man eigentlich nur ne Trassenfinder-Route angeben muss und dann zu den Strecken(segmenten) die vMax und den twistingFactor.

Haltestellen werden vollautomatisch mit allen Daten hinzugefügt.
Bei den Bahnhofskategorien habe ich das so eingeteilt, dass Preisklasse 1-2 Knotenpunkte sind, PK 3 Hauptbahnhof, Rest Kleinbahnhof

**Reply by C1710** [reply]
Nur sind die automatischen Koordinaten so halt etwas durcheinander mit den bisherigen Koordinaten :D

**Comment by marhei** [comment]
<img width="667" alt="Capture d’écran 2022-06-25 à 00 08 21" src="https://user-images.githubusercontent.com/10304183/175699302-a6586d19-6fe9-4f49-b4b1-fb3a12aa044a.png">
An sich sehr cool, dass es die richtige Koordinaten sind, nur bricht das leider an einige stellen. Vor allem am Rhein, aber auch bei Han Münde.

**Reply by C1710** [reply]
Hm, vielleicht kann man bei den entsprechenden Punkten weiterhin das bisherige System verwenden.
Ich denke, ich werde es dann zumindest für gewisse Bereiche anwenden, wo es halbwegs passt.

Das macht dann auch komplexere Erweiterungen mit vielen Bahnhöfen deutlich einfacher.

**Reply by C1710** [reply]
> Hm, vielleicht kann man bei den entsprechenden Punkten weiterhin das bisherige System verwenden.
> Ich denke, ich werde es dann zumindest für gewisse Bereiche anwenden, wo es halbwegs passt.
> 
> Das macht dann auch komplexere Erweiterungen mit vielen Bahnhöfen deutlich einfacher.

Ich hatte das zunächst mit Osnabrück — Rheine ausprobiert, wo es halt nicht gepasst hat. Aber ich denke, so ab dem nördlichen NRW nach Norden hin könnte ich es umsetzen, weil da (auch in echt) das Netz nicht ganz so dicht ist :D

**Reply by marhei** [reply]
Ja, mir gefällt das schon sehr gut. Am Rhein würde ich halt die zwei Strecken etwas aneinander ziehen und ein paar andere Bahnhöfe ebenfalls weiter voneinander entfernt platzieren als in echt.


Zusätzlich muss ich ein bisschen an der Karte arbeiten um die Darstellung an sich zu verbessern.

**Reply by marhei** [reply]
Gerade auch unten im Saarland gefällt mir das so sehr gut.
